dns.go

```diff
@@ -168,3 +93,110 @@ func NewDNSServer(hostname, nameserver, listen string) *DNSServer {
 		nameserver: nameserver,
 	}
 }
+
+func (d *DNSServer) handleDNSRequest(authority dns.RR, udpListen *net.UDPConn, b []byte, addr *net.UDPAddr) {
```

Comment:
That's an improvement, but this is still a huge method.

I was taught that a function should fit inside one computer screen. 
While we shouldn't abide to such absolutisms, such huge functions still scream "refactor me".

Reply:
If though out GO's stuiped errors checks after each call, it will fit half of screen.